### PR TITLE
Changed Tile, set query begin date off end date if provided.

### PR DESCRIPTION
**Pull request by wasabibob**
Updated to always set the XLD task query begin date  minus X number of days.  Also removed the exception for number of days being out of range 1-90 days.  If a number larger than 90 is provided for number of days, the value is truncated to 90 days.

### CODE REVIEW

**Comment by wasabibob** [comment]
the codacy/pr finding is not really an issue, the synthetic.xml always sets a value for numberOfDays.

File: src/main/resources/xlr_xldeploy/XLDVersionsTile.py [hunk 1]
```diff
@@ -12,26 +12,29 @@
...
 
 import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
 
 if not xldeployServer:
     raise Exception("XL Deploy server ID must be provided")
 
-today = org.joda.time.DateTime()
+#set the acceptable range of values for numberOfDays
 day_range = range(1,91)
```
Comment (jdewinne) [inline src/main/resources/xlr_xldeploy/XLDVersionsTile.py]: Shouldn't this 90 or 91 be a hidden property in the `synthetic.xml`?
So containing: `hidden="true"`

Comment (wasabibob) [reply src/main/resources/xlr_xldeploy/XLDVersionsTile.py]: I can do that, was confused as to if it just not throw error on the range or if you wanted it hidden.  I can update to have hidden="true"

Comment (jdewinne) [reply src/main/resources/xlr_xldeploy/XLDVersionsTile.py]: I think you misunderstood. numberOfDays is the number the end user can enter.
maxRange would be an additional property that the admin can configure as a hidden property.

**Review by jdewinne — APPROVED** [review]

**Review by hierynomus — APPROVED** [review]

**Pull request merged**